io-sim/src/Control/Monad/IOSimPOR/Types.hs

```diff
@@ -64,8 +64,17 @@ statusWriteEffects tids = mempty{effectStatusWrites = tids}
 someTvarId :: SomeTVar s -> TVarId
 someTvarId (SomeTVar r) = tvarId r
 
+-- | Make sure we only have read effects
+--
+-- It seems we need to ignore statusWrites effects to avoid generating bad
+-- schedules. See issue #76 for examples.
```

Comment:
We must ignore the `statusWrite` effects.   This function is used to only limit the blocking steps which are blocked on `STM`.

Maybe this function should be written as:
```
onlyReadEffect e = effectWrite e == mempty
```
but I am not 100% sure.

Reply:
If you update the comment we can merge it.

Reply:
@bolt12 will the `onlyReadEffect` that I suggest above fix the test case which you mentioned here: https://github.com/input-output-hk/io-sim/issues/76#issuecomment-1495946252

Reply:
The fix I have fixes the test case mentioned in #76 . I am going to try your suggestion